python/packages/aisearch/agent_framework_aisearch/_search_provider.py

```diff
@@ -0,0 +1,785 @@
+      Use only for complex queries requiring cross-document reasoning.
+
+    Examples:
+        Using Settings class with environment variables:
+
+        .. code-block:: python
+
+            from agent_framework import ChatAgent
+            from agent_framework_azure_ai import AzureAIAgentClient
+            from agent_framework.azure import AzureAISearchContextProvider, AzureAISearchSettings
+            from azure.identity.aio import DefaultAzureCredential
+
+            # Load from environment variables
+            settings = AzureAISearchSettings()
+            search_provider = AzureAISearchContextProvider(settings=settings, credential=DefaultAzureCredential())
+
+        Semantic hybrid search with API key:
+
+        .. code-block:: python
+
+            # Direct API key string (converted to AzureKeyCredential automatically)
+            search_provider = AzureAISearchContextProvider(
+                endpoint="https://mysearch.search.windows.net",
+                index_name="my-index",
+                credential="my-api-key",  # String converted to AzureKeyCredential
+                mode="semantic",
+            )
+
+        Agentic retrieval for complex queries:
+
+        .. code-block:: python
+
+            # Use agentic mode for multi-hop reasoning (slower)
+            search_provider = AzureAISearchContextProvider(
+                endpoint="https://mysearch.search.windows.net",
+                index_name="my-index",
+                credential=DefaultAzureCredential(),
+                mode="agentic",  # Multi-hop reasoning
+                azure_ai_project_endpoint="https://myproject.services.ai.azure.com",
+                model_deployment_name="gpt-4o",
+                knowledge_base_name="my-knowledge-base",  # Required for agentic mode
+            )
+    """
+
+    _DEFAULT_SEARCH_CONTEXT_PROMPT = "Use the following context to answer the question:"
+
+    def __init__(
+        self,
+        endpoint: str | None = None,
+        index_name: str | None = None,
+        credential: str | AzureKeyCredential | AsyncTokenCredential | None = None,
+        *,
+        settings: AzureAISearchSettings | None = None,
+        mode: Literal["semantic", "agentic"] = "semantic",
+        top_k: int = 5,
+        semantic_configuration_name: str | None = None,
+        vector_field_name: str | None = None,
+        embedding_function: Callable[[str], Awaitable[list[float]]] | None = None,
+        context_prompt: str | None = None,
+        # Agentic mode parameters (Knowledge Base)
+        azure_ai_project_endpoint: str | None = None,
+        model_deployment_name: str | None = None,
+        model_name: str | None = None,
+        knowledge_base_name: str | None = None,
+        retrieval_instructions: str | None = None,
+        azure_openai_api_key: str | None = None,
+        azure_openai_resource_url: str | None = None,
+        # Deprecated parameters (for backwards compatibility)
```

Comment:
to what are we backward compatible here? maybe just remove them now?

Reply:
removed